### PR TITLE
ENH: Changed base image for singularity-shim to the 'slim' docker ima…

### LINKED ISSUE #26: singularity-shim docker image is too large
I see it as 
```
$> docker images | grep singularity-shim
mjtravers/singularity-shim                                               latest                             23ed220c0cd1        2 weeks ago         919MB
mjtravers/singularity-shim                                               <none>                             16e22847f322        5 weeks ago         919MB
mjtravers/singularity-shim                                               <none>                             6a7d567c53eb        6 weeks ago         919MB
```
so 900MB. original base images on https://hub.docker.com/r/singularityware/singularity/tags standard 356MB and there is some  -slim of 65MB.  Worth checking what is the difference -- may be -slim would be good enough for us.  But I also wonder why I have 919MB image ;)  any ideas @mjtravers ?

### PR DESCRIPTION
**Pull request by mjtravers**
…ge of singularity

Fixes #26 

@yarikoptic 

The base Docker image for the singularity-shim was changed to the "slim" version.

Also, for image stability, an exact image version is used for the base image rather than the "latest".

The new version of the singularity-shim image is 68MB compressed in the hub and 140MB uncompressed after it is pulled locally.

The new smaller image has been pushed to Docker Hub.

### CODE REVIEW

**Comment by yarikoptic** [comment]
Great.  So what is the difference between -slim and regular version?

File: scripts/Dockerfile.singularity-shim [hunk 1]
```diff
@@ -10,9 +12,9 @@ RUN printf '#!/bin/bash\n\
                 GROUPNAME=host-user\n\
                 addgroup -g $GID host-user\n\
             fi\n\
-            adduser -u $UID -G "$GROUPNAME" -H -D host-user\n\
+            adduser -u $UID -G "$GROUPNAME" -D host-user\n\
```
Comment (yarikoptic) [inline scripts/Dockerfile.singularity-shim]: what was -H for originally?

**Comment by yarikoptic** [comment]
all green now, will merge

File: scripts/Dockerfile.singularity-shim (hunk 1 above)
Comment (mjtravers) [reply scripts/Dockerfile.singularity-shim]: @yarikoptic 
The -H switch tells adduser to not make a home directory. The thought was to have minimal impact on the file system. However, v3.3.0 of Singularity wants to create a ~/.singularity directory and is demanding a home directory for the user exist.

Comment (mjtravers) [reply scripts/Dockerfile.singularity-shim]: @yarikoptic 
The slim release simply has all the "convenience" packages removed. They are both Alpine 3.10.1, but slim has had basic packages such as bash and sudo removed. Slim is absolutely just the essential packages necessary to get singularity working.

Comment (yarikoptic) [reply scripts/Dockerfile.singularity-shim]: Thanks for the explanations!
re image & apk -- please add `&& rm -rf /var/cache/apk/*` at the end that `apk update` so we strip away packages lists etc in that layer

**Pull request merged**